synapse/lib/cell.py

```diff
@@ -1156,6 +1172,27 @@ async def _bumpCellVers(self, name, updates):
 
             curv = vers
 
+    async def _runMinSpaceLoop(self):
+
+        while not self.isfini:
+            disk = shutil.disk_usage(self.dirn)
+            if (disk.free / disk.total) <= self.minspace:
+
+                await self._setReadOnly()
+
+                nexsroot = self.getCellNexsRoot()
+                nexsroot.readonly = True
+
+                free = disk.free / disk.total * 100
+                mesg = f'Free space on {self.dirn} below minimum threshold (currently {free:.2f}%)'
+                raise s_exc.LowSpace(mesg=mesg, dirn=self.dirn)
```

Comment:
since we are raising here, we're leaving this coroutine and the cell in a unrecoverable state. If someone were to free disk space, they would still need to restart their cell to make it work again?

Reply:
Yeah I was thinking that this was a somewhat bad state where the usual solution would be resizing the disk and rebooting the container

Reply:
Someone could also be running on bare metal and need to, for example, go and prune stuff from `/var/log` for example.

Reply:
Yeah we can have it keep checking and re-enable writes